scripts/bootstrap.py

```diff
@@ -0,0 +1,355 @@
+# Copyright 2023 Cognite AS
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+from typing import Optional
+
+import yaml
+from cognite.client import CogniteClient
+from cognite.client.data_classes.iam import Group
+from rich import print
+from rich.columns import Columns
+from rich.panel import Panel
+
+from .utils import CDFToolConfig
+
+
+def check_auth(
+    ToolGlobals: CDFToolConfig,
+    group_id: int = 0,
+    group_file: Optional[str] = None,
+    update_group: bool = False,
+    dry_run: bool = False,
+    verbose: bool = False,
+) -> CogniteClient:
+    cluster = ToolGlobals.environ(
+        "CDF_CLUSTER",
+        None,
+        fail=False,
+    )
+    project = ToolGlobals.environ(
+        "CDF_PROJECT",
+        None,
+        fail=False,
+    )
+    token = ToolGlobals.environ(
+        "CDF_TOKEN",
+        None,
+        fail=False,
+    )
+    client_id = ToolGlobals.environ(
+        "IDP_CLIENT_ID",
+        None,
+        fail=False,
+    )
+    client_secret = ToolGlobals.environ(
+        "IDP_CLIENT_SECRET",
+        None,
+        fail=False,
+    )
+    print("[bold]Checking current service principal/application...[/]")
+    if cluster is None or len(cluster) == 0:
+        print("  [bold red]ERROR:[/]Environment variable CDF_CLUSTER must be set.")
+        ToolGlobals.failed = True
+        return
```

Comment:
```suggestion
        exit(1)
```

Reply:
I use this pattern because the caller can then decide how to handle the failure situation. Maybe not in this case, but in other cases, the same error should be treated as exit(1) or fail silently. I'm thus a bit hesitant to terminate execution from a library function.